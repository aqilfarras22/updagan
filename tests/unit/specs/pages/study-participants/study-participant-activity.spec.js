import BootstrapVue from 'bootstrap-vue';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import studyParticipantsActivityPage from '@/pages/study-participants/study-participant-activity';

const { it, describe, expect } = global;

describe('pages/study-participants/study-participant-activity.vue', () => {
  const localVue = createLocalVue();
  localVue.use(BootstrapVue);

  const activityId = 111;

  const createMountedWrapper = testMocks => {
    const getActivityResponse = {
      participant: { name: 'name' },
      payload: '{"payload": "payload"}',
      activity: {
        id: 1,
        name: 'Uniq 1',
      },
    };

    const getActivityStub = jest.fn().mockReturnValue(Promise.resolve(getActivityResponse));
    const getParticipantDataFactStub = jest.fn()
      .mockReturnValue(Promise.resolve(getActivityResponse));

    return {
      wrapper: shallowMount(studyParticipantsActivityPage, {
        localVue,
        mocks: {
          $api: {
            getActivity: getActivityStub,
            getParticipantDataFact: getParticipantDataFactStub,
          },
          $route: {
            params: {
              activityId,
            },
          },
          ...testMocks,
        },
        stubs: {
          JsonTree: { name: 'JsonTree', render: h => h('b', 'JsonTreeCmp') },
          ParticipantInfo: { name: 'ParticipantInfo', render: h => h('b', 'ParticipantInfoCmp') },
          ActivityInfo: { name: 'ActivityInfo', render: h => h('b', 'ActivityInfoCmp') },
        },
      }),
      getActivityStub,
      getParticipantDataFactStub,
      getActivityResponse,
    };
  };

  it('should create instance', () => {
    const { wrapper } = createMountedWrapper();
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  it('should have described child components', () => {
    expect(studyParticipantsActivityPage.components.ActivityInfo).toBeTruthy();
    expect(studyParticipantsActivityPage.components.ParticipantInfo).toBeTruthy();
    expect(studyParticipantsActivityPage.components.JsonTree).toBeTruthy();
  });

  it('should have all required initial data', () => {
    const { wrapper } = createMountedWrapper();
    // initial data
    expect(wrapper.vm.participantInfo).toEqual({});
    expect(wrapper.vm.activity).toEqual(null);
    expect(wrapper.vm.payload).toEqual(null);
  });

  it('should return computed property: activityId', () => {
    const { wrapper } = createMountedWrapper();
    expect(wrapper.vm.activityId).toBe(activityId);
  });

  it('should has the expected html structure', async () => {
    const { wrapper, getActivityStub, getParticipantDataFactStub } = createMountedWrapper();
    await getParticipantDataFactStub();
    await getActivityStub();
    await wrapper.vm.$nextTick();
    expect(wrapper.vm.$el).toMatchSnapshot();
  });
});
