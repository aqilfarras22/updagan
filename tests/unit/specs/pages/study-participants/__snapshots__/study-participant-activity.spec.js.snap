// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/study-participants/study-participant-activity.vue should has the expected html structure 1`] = `
<b-container-stub
  class="study-particip-activity-page pt-30"
  fluid="true"
  tag="div"
>
  <b-row-stub
    tag="div"
  >
    <b-col-stub
      cols="2"
      tag="div"
    >
      <b
        id="1"
        name="Uniq 1"
      >
        ActivityInfoCmp
      </b>
       
      <hr
        class="mt-30 mb-30"
      />
       
      <b>
        ParticipantInfoCmp
      </b>
    </b-col-stub>
     
    <b-col-stub
      tag="div"
    >
      <b-card-stub
        bodytag="div"
        footertag="div"
        headertag="div"
        subtitletag="h6"
        tag="div"
        titletag="h4"
      >
        <b
          json="[object Object]"
          readablekeys=""
        >
          JsonTreeCmp
        </b>
      </b-card-stub>
    </b-col-stub>
  </b-row-stub>
</b-container-stub>
`;
